### DISCUSSION TITLE
send messages to channel inside a super group.

### DISCUSSION THREAD

**Discussion by AguChaves92** [telegraf/telegraf#1811] [Q&A] [answered]
So my question is how do you send a message to a specific channel inside a supergruop. 

`
bot.on("text", (ctx) => {
  console.log(ctx.chat);
  bot.telegram.sendMessage("-11111111111", "hi");
});`

but this makes the bot sent the message to the general channel. not the one i want it to. 
I've tried to get the channel id by using ctx.chat.id, but that returns the group's id and not the channel's one. Also I've found a prop called message_thread_id that has a number that corresponds with the number the channel gets after the normal url :"common_channel:link/3"
does this have something to do with it?

**Comment by MKRhere** [comment, marked as answer]
You mean specific topic in a supergroup?

The shorthand for "send a message in the same group and topic" is:

```TS
ctx.sendMessage("hi");
```

does this automatically. If you have to use the long form, it's:

```TS
const topic = ctx.message.is_topic_message ? ctx.message.message_thread_id : undefined;
bot.telegram.sendMessage(ctx.chat.id, "hi", { message_thread_id: topic });
```

**Reply by AguChaves92** [reply]
thx

**Comment by pooriaarab** [comment]
@MKRhere How can we send Telegram typing action to a specific topic in supergroup?

Right now, telegram returns ChatID which is for the whole group and I'm sending typing action to that ChatID but it is being sent to #general topic instead of #specific_topic. 

Any idea if that is possible in Telegram bot API?

**Reply by MKRhere** [reply]
Have you upgraded Telegraf? In latest Telegraf, `ctx.reply` and `ctx.sendMessage` automatically send the message in the current topic. If you need to send a message to a specific topic other than the current, and you need to use `bot.telegram.sendMessage`, you must pass the `message_thread_id` of that topic.

**Reply by MKRhere** [reply]
See my other reply in this thread for more details.

**Comment by pooriaarab** [comment]
Thx for the quick response. 

I'm using https://stackoverflow.com/a/75178418

And basically passing it as {chat_id}/{message_thread_id} but still the typing action is being sent to the #general in group not the #specific_topic in supergroup

**Reply by MKRhere** [reply]
> And basically passing it as {chat_id}/{message_thread_id}

You cannot do this with the Bot API. Show your real code so I can suggest a fix.
